### PR TITLE
fix: capture a $pageview event on opting in

### PR DESCRIPTION
**Pull request by skoob13**
## Changes

The initial `$pageview` is not captured when customers use the opt in/out flow. It creates a discrepancy in event count like [this](https://posthog.slack.com/archives/C0113360FFV/p1723716455610139) one. The PR changes the behavior to capture the initial `$pageview` event after a user opts in if the event hasn't yet been sent.

**Before**

Screenshot 2024-08-20 at 18 01 59

**After**

Screenshot 2024-08-20 at 18 12 59

## Checklist
- [x] Tests for new code (see [advice on the tests we use](https://github.com/PostHog/posthog-js#tiers-of-testing))
- [x] Accounted for the impact of any changes across different browsers
- [x] Accounted for backwards compatibility of any changes (no breaking changes in posthog-js!)

### CODE REVIEW

File: src/posthog-core.ts [hunk 1]
```diff
@@ -1977,12 +1980,15 @@ export class PostHog {
...
+        // Don't capture if captureEventName is null or false
+        if (isUndefined(options?.captureEventName) || options?.captureEventName) {
+            this.capture(options?.captureEventName ?? '$opt_in', options?.captureProperties, { send_instantly: true })
         }
 
-        this.capture(options?.captureEventName ?? '$opt_in', options?.captureProperties, { send_instantly: true })
+        if (this.config.capture_pageview && !this._initialPageviewCaptured && document) {
+            this._initialPageviewCaptured = true
+            this.capture('$pageview', { title: document.title }, { send_instantly: true })
+        }
```
Comment (benjackwhite) [inline src/posthog-core.ts:1991]: Could we move this block into a private method like `captureInitialPageview` instead. that way it isn't likely to get forgotten about later in one place or the other

**Review by benjackwhite — CHANGES_REQUESTED** [review]
Generally looks good. Just a change to make it easier to maintain in the future

File: src/posthog-core.ts (hunk 1 above)
Comment (skoob13) [reply src/posthog-core.ts]: Thanks @benjackwhite! I added a private method.

**Review by benjackwhite — APPROVED** [review]

**Comment by benjackwhite** [comment]
Approved - dont forget to bump the label

**Pull request merged**